Mathlib/Algebra/Category/ContinuousCohomology/Basic.lean

```diff
@@ -0,0 +1,159 @@
+
+namespace ContinuousCohomology
+
+variable [TopologicalSpace G] [IsTopologicalGroup G]
+
+variable {R G} in
+/-- The `G` representation `C(G, rep)` given a representation `rep`.
+The `G` action is defined by `g • f := x ↦ f (g⁻¹ * x)`. -/
+abbrev Iobj (rep : Action (TopModuleCat R) G) : Action (TopModuleCat R) G where
+  V := .of R C(G, rep.V)
+  ρ :=
+  { toFun g := TopModuleCat.ofHom
+      { toFun f := .comp (rep.ρ g).hom (f.comp (Homeomorph.mulLeft g⁻¹))
+        map_add' _ _ := by ext; simp
+        map_smul' _ _ := by ext; simp
+        cont := ((continuous_postcomp _).comp (continuous_precomp _)) }
+    map_one' := ConcreteCategory.ext (by ext; simp)
+    map_mul' _ _ := ConcreteCategory.ext (by ext; simp [mul_assoc]) }
+
+/-- The functor taking a representation `rep` to the representation `C(G, rep)`. -/
+@[simps]
+def I : Action (TopModuleCat R) G ⥤ Action (TopModuleCat R) G where
+  obj := Iobj
+  map {M N} φ :=
+  { hom := TopModuleCat.ofHom (ContinuousLinearMap.compLeftContinuous _ _ φ.hom.hom)
+    comm g := by
+      ext f g'
+      show (M.ρ g ≫ φ.hom).hom (f (g⁻¹ * g')) = (φ.hom ≫ N.ρ g).hom (f (g⁻¹ * g'))
+      rw [φ.comm] }
+  map_id _ := rfl
+  map_comp _ _ := rfl
+
+instance : (I R G).Additive := ⟨by tauto⟩
+instance : (I R G).Linear R := ⟨by tauto⟩
+
+/-- The constant function `rep ⟶ C(G, rep)` as a natural transformation. -/
+@[simps]
+def const : 𝟭 _ ⟶ I R G where
+  app _ := { hom := TopModuleCat.ofHom (.const _ _), comm _ := rfl }
+  naturality _ _ _ := rfl
+
+namespace MultiInd
+
+/-- The n-th functor taking `M` to `C(G, C(G,...,C(G, M)))` (with n `G`s).
+These functors form a complex, see `MultiInd.complex`. -/
+def functor : ℕ → Action (TopModuleCat R) G ⥤ Action (TopModuleCat R) G
+  | 0     => 𝟭 _
+  | n + 1 => functor n ⋙ I R G
+
+/-- The differential map in `MultiInd.complex`. -/
+def d : ∀ n : ℕ, functor R G n ⟶ functor R G (n + 1)
+  | 0     => const R G
+  | n + 1 => whiskerLeft (functor R G (n + 1)) (const R G) - (by exact whiskerRight (d n) (I R G))
+
+lemma d_zero : d R G 0 = const R G := rfl
+
+lemma d_succ (n : ℕ) :
+    d R G (n + 1) = whiskerLeft (functor R G (n + 1)) (const R G) -
+      (by exact whiskerRight (d R G n) (I R G)) := rfl
+
+@[simp]
+lemma whiskerRight_zero {C D E : Type*} [Category C] [Category D] [Category E]
+    {F G : C ⥤ D} (H : D ⥤ E) [Limits.HasZeroMorphisms D] [Limits.HasZeroMorphisms E]
+    [H.PreservesZeroMorphisms] :
+    whiskerRight (0 : F ⟶ G) H = 0 := by aesop_cat
+
+@[reassoc (attr := simp)]
+lemma d_comp_d (n : ℕ) :
+    d R G n ≫ d R G (n + 1) = 0 := by
+  induction n with
+  | zero =>
+    rw [d_succ, Preadditive.comp_sub, sub_eq_zero]
+    rfl
+  | succ n ih =>
+    rw [d_succ R G (n + 1), Preadditive.comp_sub]
+    nth_rw 2 [d_succ]
+    rw [Preadditive.sub_comp, ← whiskerRight_comp, ih, whiskerRight_zero, sub_zero, sub_eq_zero]
+    rfl
+
+/-- The complex of functors taking `M` to `C(G, C(G,...,C(G, M)))`. -/
```

Comment:
I think that this is the complex of functors taking `M` to the complex `M -> C(G, M) -> C(G, C(G, M)) -> ...`.